### PR TITLE
[behavioural_qc] Testplan revamp

### LINKED ISSUE #7193: [Behaviour Quality Control] Update Test Plan
Issue originally reported by @HenriRabalais on CCNA Repo during candidate release testing of `LORIS 23.0.2`
aces/CCNA#4259

_____

There are certain elements of the Test Plan that are unclear or not specific which slows down the testing process and makes it more prone to error.

**First Issue**
Point 3 states: `Remove the 'access all profiles' permission and test whether the site filter is now populated only with sites that are study sites and to which the user has access.`

There is no permission that is called `access all profiles`. I believe the permission that is being referred to is called `Across all sites access candidate profiles`

**Second Issue**
Point 5 states: `Ensure that Conflict/Incomplete Candidate/Link to BVL Feedback links point to the correct place.`

It is unclear what each of these listed items refer to, as they do not match the tab labels within the module. There are also many links that can be clicked on, and the testing plan point does not state what link should be tested.

**Third Issue**
Point 6 states: `Ensure that feedback status is correct.`

It is unclear what `feedback status` refers to in the context of this module, since the word `status` doesn't appear in the module.

**Fourth Issue**
Point 9 states: `Perform a query that returns a row with a link to BVL feedback. Click on the click to edit the feedback QC status and change its value. Save the data and close the popup. Make sure that the behavioural_qc result table has been updated for that row (i.e. reflect the new Feedback status). Click on the feedback link again and check that the QC status has also been updated in the popup window.`

Either I can't find it, or the ability to edit through a pop-up window is inaccessible or hidden. Whether it exists or not, this instruction is rather unclear and should probably be rewritten.

**N.B.** 
This may be coming from the perspective of someone who doesn't use the module, and this may be much clearer to a regular user, but I had a lot of trouble following these instructions!

### PR DESCRIPTION
**Pull request by ridz1208**
## Brief summary of changes
Multiple changes to the format/content of the testplan

***NOTE:** To whomever is testing this. The module currently has a bug where data returned is NOT filtered for the user's sites so the module in its current state will fail this test plan but that issue is being resolved in a different PR.*
#### Link(s) to related issue(s)

* Resolves #7193

### CODE REVIEW

File: modules/behavioural_qc/test/TestPlan.md [hunk 1]
```diff
@@ -1,20 +1,15 @@
...
-7. Repeat steps 2-8 with applying all the filters.
-8. Perform a query that returns a row with a link to BVL feedback. Click on the click to edit the feedback QC
-   status and change its value. Save the data and close the popup. Make sure that the behavioural_qc result table has been 
-   updated for that row (i.e. reflect the new Feedback status). Click on the feedback link again and check that 
-   the QC status has also been updated in the popup window.  
+1. Make sure the user can only access the module if they have the permission 
+_Behavioural Quality Control: View Flagged Behavioural Entries_ (`behavioural_quality_control_view`)
+2. Make sure that without the _Access Profile: View Candidates and
+Timepoints - All Sites_ (`access_all_profiles`) permission the user only has 
+access to data affilioated with their site and with the permission they can see data
```
Comment (cmadjar) [inline modules/behavioural_qc/test/TestPlan.md:7]:
```suggestion
access to data affiliated with their site and with the permission they can see data 
```

File: modules/behavioural_qc/test/TestPlan.md [hunk 2]
```diff
@@ -1,20 +1,15 @@
...
-   updated for that row (i.e. reflect the new Feedback status). Click on the feedback link again and check that 
-   the QC status has also been updated in the popup window.  
+1. Make sure the user can only access the module if they have the permission 
+_Behavioural Quality Control: View Flagged Behavioural Entries_ (`behavioural_quality_control_view`)
+2. Make sure that without the _Access Profile: View Candidates and
+Timepoints - All Sites_ (`access_all_profiles`) permission the user only has 
+access to data affilioated with their site and with the permission they can see data 
+from all sites.
+3. Make sure the 3 tabs load _Incomplete Forms_, _Data Conflicts_ and _Behavioural Feedback_.
+4. Make sure links on all page are function and redirect the user to the appropriate page.
```
Comment (cmadjar) [inline modules/behavioural_qc/test/TestPlan.md:10]:
```suggestion
4. Make sure links on all pages are functional and redirect the user to the appropriate page. 
```

File: modules/behavioural_qc/test/TestPlan.md [hunk 3]
```diff
@@ -1,20 +1,15 @@
...
+_Behavioural Quality Control: View Flagged Behavioural Entries_ (`behavioural_quality_control_view`)
+2. Make sure that without the _Access Profile: View Candidates and
+Timepoints - All Sites_ (`access_all_profiles`) permission the user only has 
+access to data affilioated with their site and with the permission they can see data 
+from all sites.
+3. Make sure the 3 tabs load _Incomplete Forms_, _Data Conflicts_ and _Behavioural Feedback_.
+4. Make sure links on all page are function and redirect the user to the appropriate page. 
+Links to PSCID/DCCID should redirect to the candidate page, visits to the candidate's 
+timepoint, Instruments to the data entry form and feedback links to the appropriate 
+location where the feedback was entered.
```
Comment (cmadjar) [inline modules/behavioural_qc/test/TestPlan.md:13]:
```suggestion
Links to PSCID/DCCID should redirect to the candidate page. Links to visits  should redirect to the 
candidate's timepoint. Links to instruments should redirect to the data entry form. Finally, feedback links 
should open the appropriate window where the feedback was entered.
```

I might be blind but I cannot find the feedback links. Where are they? If I click on feedback status, it does not do anything...

**Review by cmadjar — CHANGES_REQUESTED** [review]
@ridz1208 a few minor changes to the text and one question :)

File: modules/behavioural_qc/test/TestPlan.md (hunk 3 above)
Comment (ridz1208) [reply modules/behavioural_qc/test/TestPlan.md]: I think the "Feedback level" links used to open the instrument/visit/candidate with the bvl_feedback window open to see the feedback directly

https://github.com/aces/Loris/issues/8056

**Comment by ridz1208** [comment]
@cmadjar sorry for the million typos

File: modules/behavioural_qc/test/TestPlan.md (hunk 3 above)
Comment (cmadjar) [reply modules/behavioural_qc/test/TestPlan.md]: aaaaah, ok. Thanks for the clarification.

**Review by cmadjar — APPROVED** [review]
👍

**Pull request merged**